docs/dogstatsd/internals.md

```diff
@@ -3,66 +3,92 @@
-Once fully assembled by the PacketAssembler, the bytes array is sent through the rest of the Dogstatsd pipeline and ownership is allocated to each part using it (PacketsBuffer, Worker). Eventually, the Worker takes care of returning it to the pool when the part has processed its content.
+Once fully assembled by the PacketAssembler, the bytes array is sent through the rest of the DogStatsD pipeline and ownership is allocated to each part using it (PacketsBuffer, Worker). Eventually, the Worker takes care of returning it to the pool when the part has processed its content.
 
 ## PacketBuffer
 
-Input: a Packet containing one or several metrics in the Dogstatsd format (separated by a \n)
+Input: a Packet containing one or several metrics in the DogStatsD format (separated by a \n)
 Output: multiple Packets send in a row to the Worker
 
 The PacketsBuffer buffers multiple Packets (in a slice), this way the parsing part of the pipeline is going through several Packets in a row instead of only one each time it is called. This leads to less CPU usage. PacketsBuffer sends the Packets for processing when either:
+
 * The buffer is full (contains `dogstatsd_packet_buffer_size, default value: 32`)
+
 or
 
 * A timer is triggered (i.e. `dogstatsd_packer_buffer_flush_timeout, default value: 100ms`)
 
 The PacketBuffer sends it in a Go buffered channel to the worker / parser, meaning that the channels can buffer the Packets on their own while waiting for the worker to read and process them.
 
 In theory, the max memory usage of this Go buffered channel is:
-* packet buffer size * packet size * channel buffer size
-* `dogstatsd_packer_buffer_size` * `dogstatsd_buffer_size` * `dogstatsd_queue_size`
-* 32 * 8192 * 1024 =  256MB
+
+  * packet buffer size * packet size * channel buffer size
+  * `dogstatsd_packer_buffer_size` * `dogstatsd_buffer_size` * `dogstatsd_queue_size`
+  * 32 * 8192 * 1024 =  256MB
 
 ## Worker
 
 Input: slice of Packets
 
 Output: MetricSample sent
 
-The Worker is the part of the Dogstatsd server responsible for parsing the metrics in the bytes array and turning them into MetricSamples. The server spawns multiple workers based on the amount of cores available on the host (the amount of workers created is equal to the number of cores on the machine minus 2. If this result is less than 2, 2 workers are spawned).
+The Worker is the part of the DogStatsD server responsible for parsing the metrics in the bytes array and turning them into MetricSamples.
+
+The server spawns multiple workers based on the amount of cores available on the host:
+
+  * when we are not sharding multiple time sampling pipelines: the amount of workers created is equal to the number of cores on the machine minus 2. If this result is less than 2, 2 workers are spawned.
+  * when we are: the amount of workers is `(number of cores / 2)`.  If this result is less than 2, 2 workers are spawned.
```

Comment:
- Please capitalize the first character of each list item.
- Please rewrite these sentences so they don't use "we". I'm not able to rewrite it myself, because I don't know what entity is doing the sharding. Is it the DogStatsD server, perhaps? In that case, you could replace "we are" with "the server is".
- Also please make sure that all list items are complete sentences. The second item should probably look something like this: "When the server shards multiple time sampling pipelines, the server creates `(number of cores / 2 )` workers. If this result is less than 2, the server spawns 2 workers"